### PR TITLE
WIP add diagnostics bundle rotation

### PR DESCRIPTION
**Pull request by mnaboka**

### CODE REVIEW

**Comment by pcrews** [comment]
test this please

**Comment by pcrews** [comment]
test this please

**Comment by pcrews** [comment]
test this please

**Comment by pcrews** [comment]
test this please

**Comment by pcrews** [comment]
test this please

File: api/diagnostics.go [hunk 1]
```diff
@@ -22,6 +22,10 @@ import (
 	"github.com/dcos/dcos-diagnostics/util"
 	"github.com/dcos/dcos-go/exec"
 	"github.com/shirou/gopsutil/disk"
+	"regexp"
+	"google.golang.org/genproto/googleapis/type/date"
+	"strconv"
+	"sort"
```
Comment (janisz) [inline api/diagnostics.go]: Please sort imports. You can use [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports)

```diff
api/diagnostics.go
diff -u api/diagnostics.go.orig api/diagnostics.go
--- api/diagnostics.go.orig	2018-12-29 01:24:24.664985289 +0000
+++ api/diagnostics.go	2018-12-29 01:24:24.664985289 +0000
@@ -22,10 +22,10 @@
 	"github.com/dcos/dcos-diagnostics/util"
 	"github.com/dcos/dcos-go/exec"
 	"github.com/shirou/gopsutil/disk"
-	"regexp"
 	"google.golang.org/genproto/googleapis/type/date"
-	"strconv"
+	"regexp"
 	"sort"
+	"strconv"
 )
```

File: api/diagnostics.go [hunk 2]
```diff
@@ -328,6 +332,85 @@ func (j *DiagnosticsJob) runBackgroundJob(nodes []Node, cfg *config.Config, DCOS
...
+	if len(errors) > 0 {
+		return nil, fmt.Errorf("error removeing bundles: %s", strings.Join(errors, "; "))
+	}
+
+	return rotatedFiles, nil
+}
+
+// rotateBundles keeps last n bundles on the filesystem and removes the outdated ones
+func (j *DiagnosticsJob) rotateBundles(n int, listBundlesFn func(path string) ([]os.FileInfo, error),
+		rmFn func(bundlePath string) error) (removedBundles []string, err error) {
```
Comment (janisz) [inline api/diagnostics.go]: Wrong indentation run `gofmt`

**Comment by janisz** [comment]
It's stale closing

**Pull request closed without merging**